sdk/storage/azure-storage-queue/src/main/java/com/azure/storage/queue/QueueAsyncClient.java

```diff
@@ -776,16 +961,69 @@ public HttpPipeline getHttpPipeline() {
         Function<String, Mono<PagedResponse<PeekedMessageItem>>> retriever =
             marker -> StorageImplUtils.applyOptionalTimeout(this.client.getMessages()
                 .peekWithResponseAsync(queueName, maxMessages, null, null, context), timeout)
-                .map(response -> new PagedResponseBase<>(response.getRequest(),
-                    response.getStatusCode(),
-                    response.getHeaders(),
-                    response.getValue(),
-                    null,
-                    response.getDeserializedHeaders()));
+                .flatMap(this::transformMessagesPeekResponse);
 
         return new PagedFlux<>(() -> retriever.apply(null), retriever);
     }
 
+    private Mono<PagedResponseBase<MessagesPeekHeaders, PeekedMessageItem>> transformMessagesPeekResponse(
+        MessagesPeekResponse response) {
+        List<PeekedMessageItemInternal> peekedMessageInternalItems = response.getValue();
+        if (peekedMessageInternalItems == null) {
+            peekedMessageInternalItems = Collections.emptyList();
+        }
+        return Flux.fromIterable(peekedMessageInternalItems)
+            .flatMap(peekedMessageItemInternal ->
```

Comment:
`flatMap` can potentially mess up the order of peeked messages. If the order is important, use `flatMapSequential` or `concatMap` instead.